#54 tooltips in filters

sergey-zakharoff opened the pull request:
Draft implementation for 
https://github.com/morpher-ru/gramdict/issues/54
Added tooltips for search filters.
![filter-tooltips](https://user-images.githubusercontent.com/50071892/56964816-97408380-6b64-11e9-9fe7-f3610ab4b35e.png)

bzaar commented:
Thanks! Looks good, for a draft! 

Замечания:

* На скриншоте выдающаяся часть тултипа указывает скорее на "част.", чем на "предик." Нужно поправить позиционирование тултипа. 
* В тултипе должна быть еще ссылка "Буквенные символы" - см. задачу #54.

sergey-zakharoff commented:
> * В тултипе должна быть еще ссылка "Буквенные символы" - см. задачу #54.
Текст тултипа - гиперссылка на соответствующие страницы из таски. Мне показалось, что отдельные ссылки вида <Буквенные символы> в каждом тултипе перегружают вид, поэтому, после ряда экспериментов, я остановился на этом варианте. В принципе, ничего не стоит сделать и буквальный вариант для сравнения.


> * Нужно поправить позиционирование тултипа.
Ok, согласен.

bzaar commented:
Хорошо, доверяю твоему чувству прекрасного :)

Жду поправку позиции тултипа.

sergey-zakharoff commented:
fixed position and, also, implemented tooltips with related  "буквенные символы" etc labels.
Here is a couple of examples:
![filter-tooltips-2](https://user-images.githubusercontent.com/50071892/57029485-8b2ef180-6c4a-11e9-979c-a44e32cd25f0.png)
And (long tooltip)
![filter-tooltips-3](https://user-images.githubusercontent.com/50071892/57029514-a7329300-6c4a-11e9-8394-ba1d91e1b40b.png)

bzaar commented:
That looks really good, thank you! Do you think the long tooltips would
look better with a max-width of 40ex or similar?



On Wed, 1 May 2019, 17:52 sergey-zakharoff, <notifications@github.com>
wrote:

> fixed position and, also, implemented tooltips with related "буквенные
> символы" etc labels.
> Here is a couple of examples:
> [image: filter-tooltips-2]
> <https://user-images.githubusercontent.com/50071892/57029485-8b2ef180-6c4a-11e9-979c-a44e32cd25f0.png>
> And (long tooltip)
> [image: filter-tooltips-3]
> <https://user-images.githubusercontent.com/50071892/57029514-a7329300-6c4a-11e9-8394-ba1d91e1b40b.png>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/morpher-ru/gramdict/pull/55#issuecomment-488340106>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA6XT6XXH2D46Z4TXIK475TPTHDFHANCNFSM4HJLS7XQ>
> .
>

sergey-zakharoff commented:
Next attempt:
![filter-tooltips-4](https://user-images.githubusercontent.com/50071892/57040933-8f1e3c00-6c69-11e9-9873-bd7e0fc4b5af.png)

bzaar commented:
Отлично, спасибо! Завтра утром постараюсь проверить, как оно работает,
запустив у себя.

On Wed, 1 May 2019, 21:34 sergey-zakharoff, <notifications@github.com>
wrote:

> Next attempt:
> [image: filter-tooltips-4]
> <https://user-images.githubusercontent.com/50071892/57040933-8f1e3c00-6c69-11e9-9873-bd7e0fc4b5af.png>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/morpher-ru/gramdict/pull/55#issuecomment-488411654>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA6XT6RSFLR4ITDRRN5IB43PTH5EDANCNFSM4HJLS7XQ>
> .
>

bzaar commented:
Почему-то *от* странным образом поменяло цвет:

![image](https://user-images.githubusercontent.com/4028922/57045180-b45c7b80-6c64-11e9-8296-bf0bf19a0e79.png)

bzaar commented:
А вот тултипов у себя не вижу, хоть убей. Может быть, не все закоммичено?

sergey-zakharoff commented:
something strange.
I cannot confirm both of the issues (wrong color and tooltip disappearing). 
I've cleared solution, rebuild/reload packeges, etc. Locally everything seems ok.

bzaar commented:
Can you try a fresh checkout?

On Thu, 2 May 2019, 09:25 sergey-zakharoff, <notifications@github.com>
wrote:

> something strange.
> I cannot confirm both of the issues (wrong color and tooltip disappearing).
> I've cleared solution, rebuild/reload packeges, etc. Locally everything
> seems ok.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/morpher-ru/gramdict/pull/55#issuecomment-488589375>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA6XT6W6NPDNB6KD7NME4P3PTKQRJANCNFSM4HJLS7XQ>
> .
>

sergey-zakharoff commented:
Did you take latest (d8eeb8688) commit?

sergey-zakharoff commented:
Ok, sure, I'll take it soon.

sergey-zakharoff commented:
> Can you try a fresh checkout?
> […](#)
> On Thu, 2 May 2019, 09:25 sergey-zakharoff, ***@***.***> wrote: something strange. I cannot confirm both of the issues (wrong color and tooltip disappearing). I've cleared solution, rebuild/reload packeges, etc. Locally everything seems ok. — You are receiving this because you commented. Reply to this email directly, view it on GitHub <[#55 (comment)](https://github.com/morpher-ru/gramdict/pull/55#issuecomment-488589375)>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA6XT6W6NPDNB6KD7NME4P3PTKQRJANCNFSM4HJLS7XQ> .

Checked. here is new screenshot from completely fresh checkout:
![filter-tooltips-5](https://user-images.githubusercontent.com/50071892/57064540-49945a00-6ccf-11e9-9cbd-62a9efa4d291.png)

Sergey, what branch did you try at my fork?

bzaar commented:
Ура! Тултипы заработали. Не знаю, что у меня до этого было. 

Две проблемы:

1. Тултипы мешают выбирать нужные фильтры. Можно добавить секундную задержку перед появлением тултипа?

2. Чтобы кликнуть на ссылку в тултипе, нужно очень сильно изловчиться, т.к тултип постоянно убегает от мышки и потом вообще пропадает. Нельзя ли сделать, чтобы он не убегал, а стоял на месте, пока курсор мыши над выбранным фильтром и еще полсекунды?

bzaar commented:
Залил этот бранч на gramdict.ru - там тоже нет тултипов :( Можешь разобраться, почему?
